vmm/src/memory_manager.rs

```diff
@@ -232,13 +240,28 @@ impl MemoryManager {
             .collect();
 
         let mut mem_regions = Vec::new();
-        for region in ram_regions.iter() {
-            mem_regions.push(MemoryManager::create_ram_region(
-                &config.file,
-                region.0,
-                region.1,
-                false,
-            )?);
+        if let Some(ext_backing_files) = &ext_backing_files {
+            if ram_regions.len() != ext_backing_files.len() {
+                return Err(Error::InvalidAmountExternalBackingFiles);
+            }
```

Comment:
I think the check here should be that `ext_backing_files.len()` must be greater or equal to `ram_regions.len()`.
If we snapshotted a VM where we hotpluged some RAM, the the number of backing files will be larger than the architectural RAM regions.

Reply:
Oh yes I completely missed this corner case! And it's not only about validation, but also we need to create the regions based on the amount of backing files, and not based on the number of RAM regions.